temporal-opentracing/src/main/java/io/temporal/opentracing/internal/SpanFactory.java

```diff
@@ -50,9 +49,10 @@ public SpanFactory(OpenTracingOptions options) {
       String workflowType,
       long startTimeMs,
       String workflowId) {
-    Map<String, String> tags = ImmutableMap.of(StandardTagNames.WORKFLOW_ID, workflowId);
-    String operationName =
-        options.getSpanOperationNamePrefix(operationType) + PREFIX_DELIMITER + workflowType;
+    StartSpanContext context =
```

Comment:
Passing multiple strings into one function is error prone, I suggest we use builder patter as I've mentioned above and make it more explicit.

Reply:
My concern is that it's easy to make a mistake and accidentally reorder parameters during refactoring and on the programmatic level, most likely you would only be able to validate set/not set, but not that what's set is actually what it's supposed to be. Builder eliminates this concern, also there are multiple places where we do pass nulls there...

Reply:
Builder created.